elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -408,19 +408,32 @@ export class PeerPool extends EventEmitter {
 		peer.send(message);
 	}
 
-	public discoverSeedPeers(): void {
-		const openOutboundSlots = this._getAvailableOutboundConnectionSlots();
+	public discoverFromSeedPeers(): void {
+		const openOutboundSlots = this.getAvailableOutboundConnectionSlots();
 
 		if (openOutboundSlots === 0 || this._peerLists.seedPeers.length === 0) {
 			return;
 		}
 
-		const seedPeersForConnection = shuffle(
+		const seedPeersForDiscovery = shuffle(
 			this._peerLists.seedPeers.slice(0, openOutboundSlots),
 		);
 
-		seedPeersForConnection.forEach(peer => {
-			this._addOutboundPeer(peer, this._nodeInfo as P2PNodeInfo);
+		seedPeersForDiscovery.forEach(peer => {
+			// LIP-0004 re-discovery SeedPeers when Outboundconnection < maxOutboundconnections
+			const seedPeer = this.getPeer(peer.peerId);
+			if (seedPeer) {
+				// tslint:disable-next-line: no-floating-promises
+				(async () => {
+					try {
+						await seedPeer.discoverPeers();
+					} catch (err) {
+						throw err;
+					}
+				})();
+			} else {
```

Comment:
Prefer adding outbound peer and returning early if `!seedPeer` to avoid unnecessary `else`